lib/min_heap.rb

```diff
@@ -39,33 +44,52 @@ def to_s
     end
 
     output += @store.last.value + "]"
-      
+
     return output
   end
 
   # This method returns true if the heap is empty
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(1) ? I think the length of arrays is somehow cached or something in Ruby? If not, then O(n) where n is the number of elements in the array
+  # Space complexity: O(1)
   def empty?
-    raise NotImplementedError, "Method not implemented yet..."
+    return @store.length == 0
   end
 
   private
 
   # This helper method takes an index and
-  #  moves it up the heap, if it is less than it's parent node.
+  #  moves it up the heap, if it is less than its parent node.
   #  It could be **very** helpful for the add method.
-  # Time complexity: ?
-  # Space complexity: ?
-  def heap_up(index)
-    
+  # Time complexity: O(log n) where n is the number of nodes in the heap
+  # Space complexity: O(1)
+  def heap_up(current)
+    parent = (current - 1) / 2
+    while @store[current].key < @store[parent].key do
+      @store[parent], @store[current] = @store[current], @store[parent]
+      current = parent
+      parent = current / 2
+    end
   end
 
-  # This helper method takes an index and 
-  #  moves it up the heap if it's smaller
-  #  than it's parent node.
-  def heap_down(index)
-    raise NotImplementedError, "Method not implemented yet..."
+  # This helper method takes an index and
+  #  moves it down the heap if it's larger
+  #  than its child nodes.
+  def heap_down(current)
+    current = 0
+    left_child = current * 2 + 1
+    right_child = current * 2 + 2
+    child = @store[left_child].key < @store[right_child].key ? left_child : right_child
+
+    while @store[current].key > @store[child].key do
```

Comment:
What happens if `@store[left_child]` is `nil`?  I think this could cause an error.